mne_bids/tests/test_read.py

```diff
@@ -295,19 +296,65 @@ def test_handle_coords_reading():
     write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
 
     # read in the data and assert montage is the same
+    # regardless of 'm', 'cm', 'mm', or 'pixel'
     bids_fname = bids_basename + "_ieeg.edf"
-    raw_test = read_raw_bids(bids_fname, bids_root)
-
-    # obtain the sensor positions
-    orig_locs = raw.info['dig'][1]
-    test_locs = raw_test.info['dig'][1]
-    assert orig_locs == test_locs
-    assert not object_diff(raw.info['chs'], raw_test.info['chs'])
-
-    # test error message if electrodes don't match
+    coordinate_units = ['m', 'cm', 'mm']
+    scaling_number = [1., 100., 1000.]
+    coordsystem_fname = _find_matching_sidecar(bids_fname, bids_root,
+                                               suffix='coordsystem.json')
     electrodes_fname = _find_matching_sidecar(bids_fname, bids_root,
                                               "electrodes.tsv",
                                               allow_fail=True)
+    orig_electrodes_dict = _from_tsv(electrodes_fname,
+                                     [str, float, float, float, str])
+    for i, coord_unit in enumerate(coordinate_units):
+        # update coordinate units
+        _update_sidecar(coordsystem_fname, 'iEEGCoordinateUnits', coord_unit)
+        electrodes_dict = _from_tsv(electrodes_fname,
+                                    [str, float, float, float, str])
+        for axis in ['x', 'y', 'z']:
+            electrodes_dict[axis] = np.multiply(orig_electrodes_dict[axis],
+                                                scaling_number[i])
+        _to_tsv(electrodes_dict, electrodes_fname)
+
+        # read in raw file w/ updated montage
+        raw_test = read_raw_bids(bids_fname, bids_root)
+
+        # obtain the sensor positions and make sure they're the same
+        for j in range(len(raw.info['dig'])):
+            np.testing.assert_array_equal(raw.info['dig'][j],
+                                          raw_test.info['dig'][j])
+
+    # check that coordinate systems can be used and defaults to mri
+    coordinate_frames = ['lia', 'ria', 'lip', 'rip', 'las']
+    mri_coord_frame_int = _str_to_frame["mri"]
+    for coord_frame in coordinate_frames:
+        # update coordinate units
+        _update_sidecar(coordsystem_fname, 'iEEGCoordinateSystem', coord_frame)
+        # read in raw file w/ updated coordinate frame
+        raw_test = read_raw_bids(bids_fname, bids_root)
+        for digpoint in raw_test.info['dig']:
+            assert digpoint['coord_frame'] == mri_coord_frame_int
+
+    # coordinate frames in mne-python should all map correctly
+    coordinate_frames = _str_to_frame.keys()
+    for coord_frame in coordinate_frames:
+        # these coordinate frames in mne-python are related to scalp/meg
+        if coord_frame in ['meg', 'ctf_head', 'ctf_meg', 'head', 'unknown']:
+            continue
```

Comment:
this is a very weird code construction. First we import a private dictionary then we skip half of the keys. I would suggest just explicitly writing the coordinate frames like you do above.